src/EditorFeatures/CSharpTest/CodeActions/AddNamedArguments/AddNamedArgumentsTests.cs

```diff
@@ -0,0 +1,161 @@
+@"class C { C(int arg1, int arg2) { new [|C|](1, 2); } }",
+@"class C { C(int arg1, int arg2) { new C(arg1: 1, arg2: 2); } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestIndexer()
+        {
+            await TestAsync(
+@"class C { char M(string arg1) => [|arg1|][0]; }",
+@"class C { char M(string arg1) => arg1[index: 0]; }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMissingOnArrayIndexer()
+        {
+            await TestMissingAsync(
+@"class C { int M(int[] arg1) => [|arg1|][0]; }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMissingOnConditionalArrayIndexer()
+        {
+            await TestMissingAsync(
+@"class C { int? M(int[] arg1) => arg1?[|[0]|]; }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMissingOnEmptyArgumentList()
+        {
+            await TestMissingAsync(
+@"class C { void M() => [|M|](); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMissingOnAlreadyFixed()
+        {
+            await TestMissingAsync(
+@"class C { void M(int arg) => [|M|](arg: 1); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestSingleParams()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, params int[] arg2) => [|M|](1, 2); }",
+@"class C { void M(int arg1, params int[] arg2) => M(arg1: 1, arg2: 2); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMultipleParams()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, params int[] arg2) => [|M|](1, 2, 3); }",
+@"class C { void M(int arg1, params int[] arg2) => M(arg1: 1, arg2: new int[] { 2, 3 }); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestExistingArgumentNames()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, int arg2) => [|M|](1, arg2: 2); }",
+@"class C { void M(int arg1, int arg2) => M(arg1: 1, arg2: 2); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestExistingUnorderedArgumentNames()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, int arg2, int arg3) => [|M|](1, arg3: 3, arg2: 2); }",
+@"class C { void M(int arg1, int arg2, int arg3) => M(arg1: 1, arg3: 3, arg2: 2); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestPreserveTrivia()
```

Comment:
How about tests for invocations of delegates?  Keep in mind that in VB, they can be conditional invocations.